src/DebugSceneList.js

```diff
@@ -59,39 +62,56 @@ class DebugSceneList extends Component {
       all: allScenes,
       modalVisible: false,
       selectedItem: undefined,
+      search: '', // loaded in componentDidMount
     };
 
-    this.onHideScene = () =>
-      this.setState({modalVisible: false, selectedItem: undefined});
+    this.onHideScene = () => this.setState({modalVisible: false, selectedItem: undefined});
 
     this.onPressRow = (data: RegisteredItemType) => {
       this.setState({
         modalVisible: true,
         selectedItem: data,
       });
     };
+
+    // TextInput calls this when text changed.
+    // We don't need to set the state here.
+    this.handleSearchChanged = search => {
+      if (this.props.saveSearch) {
+        AsyncStorage.setItem(SAVED_SEARCH_KEY, search);
+      }
+    };
+  }
+
+  componentDidMount() {
+    // Load saved value
+    if (this.props.saveSearch) {
+      AsyncStorage.getItem(SAVED_SEARCH_KEY).then(search => this.setState({search: search || ''}));
+    }
   }
 
   renderSceneModal() {
     return (
-      <View key={'modal1'} style={[styles.selectedComponentWrapper, this.state.selectedItem && this.state.selectedItem.wrapperStyle]}>
+      <View
+        key={'modal1'}
```

Comment:
We could use here a better name, maybe something like: `component-viewer-container-modal`